Bootloader/Bootloader/Mcal/Adc/Adc.c

```diff
@@ -1,6 +1,222 @@
+				SET_BIT(ADC_ADMUX,0);
+			break;			
+			case 4:
+				SET_BIT(ADC_ADMUX,2);
+				CLEAR_BIT(ADC_ADMUX,1);
+				CLEAR_BIT(ADC_ADMUX,0);
+			break;			
+			case 5:
+				SET_BIT(ADC_ADMUX,2);
+				CLEAR_BIT(ADC_ADMUX,1);
+				SET_BIT(ADC_ADMUX,0);
+			break;			
+			case 6:
+				SET_BIT(ADC_ADMUX,2);
+				SET_BIT(ADC_ADMUX,1);
+				CLEAR_BIT(ADC_ADMUX,0);
+			break;			
+			case 7:
+				SET_BIT(ADC_ADMUX,2);
+				SET_BIT(ADC_ADMUX,1);
+				SET_BIT(ADC_ADMUX,0);
+			break;
+			default:
+				CLEAR_BIT(ADC_ADMUX,2);
+				CLEAR_BIT(ADC_ADMUX,1);
+				CLEAR_BIT(ADC_ADMUX,0);
+			break;
+		}	
+		
+		/* Choose voltage reference according to user input */
+		switch(Param_VolageReference)
+		{
+			case 0:
+				/* AREF, Internal Vref turned off */
+				CLEAR_BIT(ADC_ADMUX,7);
+				CLEAR_BIT(ADC_ADMUX,6);
+			break;			
+			case 1:
+				/* AVCC with external capacitor at AREF pin */
+				CLEAR_BIT(ADC_ADMUX,7);
+				SET_BIT(ADC_ADMUX,6);
+			break;					
+			case 3:
+				/* Internal 2.56V Voltage Reference with external capacitor at AREF pin */
+				SET_BIT(ADC_ADMUX,7);
+				SET_BIT(ADC_ADMUX,6);
+			break;
+			default:
+				/* AREF, Internal Vref turned off */
+				CLEAR_BIT(ADC_ADMUX,7);
+				CLEAR_BIT(ADC_ADMUX,6);
+			break;
+		}
+		/* Choose pre-scaler according to user input */
+		switch(Param_PreScaler)
+		{
+			case 0:
+				/* Divider = 2 */
+				CLEAR_BIT(ADC_ADCSRA,2);
+				CLEAR_BIT(ADC_ADCSRA,1);
+				CLEAR_BIT(ADC_ADCSRA,0);
+			break;			
+			case 1:
+				/* Divider = 2 */
+				CLEAR_BIT(ADC_ADCSRA,2);
+				CLEAR_BIT(ADC_ADCSRA,1);
+				SET_BIT(ADC_ADCSRA,0);
+			break;			
+			case 2:
+				/* Divider = 4 */
+				CLEAR_BIT(ADC_ADCSRA,2);
+				SET_BIT(ADC_ADCSRA,1);
+				CLEAR_BIT(ADC_ADCSRA,0);
+			break;			
+			case 3:
+				/* Divider = 8 */
+				CLEAR_BIT(ADC_ADCSRA,2);
+				SET_BIT(ADC_ADCSRA,1);
+				SET_BIT(ADC_ADCSRA,0);
+			break;			
+			case 4:
+				/* Divider = 16 */
+				SET_BIT(ADC_ADCSRA,2);
+				CLEAR_BIT(ADC_ADCSRA,1);
+				CLEAR_BIT(ADC_ADCSRA,0);
+			break;			
+			case 5:
+				/* Divider = 32 */
+				SET_BIT(ADC_ADCSRA,2);
+				CLEAR_BIT(ADC_ADCSRA,1);
+				SET_BIT(ADC_ADCSRA,0);
+			break;			
+			case 6:
+				/* Divider = 64 */
+				SET_BIT(ADC_ADCSRA,2);
+				SET_BIT(ADC_ADCSRA,1);
+				CLEAR_BIT(ADC_ADCSRA,0);
+			break;			
+			case 7:
+				/* Divider = 128 */
+				SET_BIT(ADC_ADCSRA,2);
+				SET_BIT(ADC_ADCSRA,1);
+				SET_BIT(ADC_ADCSRA,0);
+			break;
+			default:
+				/* Divider = 2 */
+				CLEAR_BIT(ADC_ADCSRA,2);
+				CLEAR_BIT(ADC_ADCSRA,1);
+				CLEAR_BIT(ADC_ADCSRA,0);
+			break;
+		}
+		
+		/* Enable Adc */
+		SET_BIT(ADC_ADCSRA,7);
+		
+		/* Enable start conversion */
+		SET_BIT(ADC_ADCSRA,6);
+		
+		/* Enable start conversion */
+		CLEAR_BIT(ADC_ADMUX,5);
+		
+		/* Enable Interrupt Flag */
+		SET_BIT(ADC_ADCSRA,4);
+		
+		/* Enable Interrupt */
+		sei();
+		SET_BIT(ADC_ADCSRA,3);
+		
+		return TRUE;
+		
+	}
+	else
+	{
+		//Error in parameters
+		return FALSE;
+	}
+}
+
+boolean Adc_ReadValue(uint16* Param_ReturnValue)
```

Comment:
What if the user wants to use more than one channel only? If I have at least two sensors and I want to read their analog values, how can I do that within your implementation?